### DISCUSSION TITLE
Add bulk delete option for record sets

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5579] [Feature Requests]
Need an option to delete all records that are members of some recordset.
Of particular challenge would be making this operation atomic, if that is indeed desirable.

**Comment by maxpatiiuk** [comment]
This would also fix #236

**Comment by grantfitzsimmons** [comment]
Requested again today on the Discourse forum ([572](https://discourse.specifysoftware.org/t/data-clean-up-of-duplicate-records/572/4))

**Comment by benanhalt** [comment]
@maxpatiiuk  you mean it would fix #236 by letting users query the audit log and convert it to a RS to delete?

From the meeting:

Jim wants to include this in a larger discussion about using record sets as the main unit of record management and whether that should be a design objective versus some alternative. Is that correct, @beach53?

It was also noted that we have generally shied away from features that can potentially wreck a database in one click, so to speak. Maybe a warning dialog that requires the user to type in the count of the number of records being deleted. Also this could be deferred until after UI for producing backup is added? Obviously the permission system would gain a new action that controls whether this function is allowed.

**Comment by bronwyncombs** [comment]
Paul Blom @ the College of Idaho has requested a bulk delete option for records as well

**Comment by grantfitzsimmons** [comment]
Mentioned by Natalia at KU
